In [1]:
import pandas as pd
import numpy as np

In [20]:
path = r"C:\Users\tpfl1\OneDrive\Desktop\SERi\voice_kor.csv"
voice = pd.read_csv(path, encoding='cp949', index_col=False)
voice.shape

(46268, 9)

In [21]:
voice.sample(5)

RECV_CPLT_DM NPA_CL EVT_STAT_CD  EVT_CL_CD RPTER_SEX  \
19914  2022-01-21 14:22:10    충남청        종결요청        215        여성   
14423  2021-09-03 17:54:52    충남청        종결요청        215        남성   
21443  2022-03-12 13:03:14    충남청        종결요청        215        남성   
30543  2022-09-28 12:07:27    충남청        종결요청        215        여성   
43587  2020-09-23 14:06:27    대전청        종결요청        215        남성   

                          HPPN_PNU_ADDR      HPPN_X     HPPN_Y SME_EVT_YN  
19914                               NaN         NaN        NaN        NaN  
14423          충청남도 아산시 온천동(온양1동) 216-4  127.002086  36.782558        NaN  
21443  충청남도 천안시 서북구 두정동 (행정:부성1동 ) 63-6  127.146547  36.830705        NaN  
30543                               NaN         NaN        NaN        NaN  
43587          대전광역시 서구 복수동(행정:복수동) 865  127.378733  36.297180

In [22]:
voice.NPA_CL.unique()

array(['충남청', '대전청', '서울청', '세종시', '인천청', '경기남부청', '전북청', '경남청', '충북청',
       '광주청', '전남청', '부산청', '강원청', '대구청', '울산청', '경기북부청', '경북청'],
      dtype=object)

In [23]:
cn = voice.loc[voice.NPA_CL == "충남청"] # 31595
dj = voice.loc[voice.NPA_CL == "대전청"] # 11176
su = voice.loc[voice.NPA_CL == "서울청"] # 79
sj = voice.loc[voice.NPA_CL == "세종시"] # 2936
ic = voice.loc[voice.NPA_CL == "인천청"] # 14
gnc = voice.loc[voice.NPA_CL == "경기남부청"] # 175
jb = voice.loc[voice.NPA_CL == "전북청"] # 52
gn = voice.loc[voice.NPA_CL == "경남청"] # 26
cb = voice.loc[voice.NPA_CL == "충북청"] # 78
kj = voice.loc[voice.NPA_CL == "광주청"] # 14
jn = voice.loc[voice.NPA_CL == "전남청"] # 26
bu = voice.loc[voice.NPA_CL == "부산청"] # 12
kw = voice.loc[voice.NPA_CL == "강원청"] # 12
dg = voice.loc[voice.NPA_CL == "대구청"] # 16
us = voice.loc[voice.NPA_CL == "울산청"] # 9
gbc = voice.loc[voice.NPA_CL == "경기북부청"] # 27
gb = voice.loc[voice.NPA_CL == "경북청"] # 21

In [48]:
cn.sample(2)

RECV_CPLT_DM NPA_CL EVT_STAT_CD  EVT_CL_CD RPTER_SEX  \
19866  2022-01-23 19:16:37    충남청        종결요청        215        남성   
1135   2020-12-21 15:09:22    충남청        종결요청        215        여성   

                          HPPN_PNU_ADDR      HPPN_X     HPPN_Y SME_EVT_YN  
19866  충청남도 천안시 동남구 신부동 (행정:신안동 ) 354-1  127.154391  36.819217        NaN  
1135        충청남도 공주시 신관동(행정:신관동) 산9-105  127.148083  36.469575          Y

In [45]:
cn.EVT_STAT_CD.unique()

array(['종결요청', '도착'], dtype=object)

In [47]:
cn.loc[cn.EVT_STAT_CD == "도착"].info() # 28

<class 'pandas.core.frame.DataFrame'>
Int64Index: 28 entries, 10593 to 46151
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   RECV_CPLT_DM   28 non-null     object 
 1   NPA_CL         28 non-null     object 
 2   EVT_STAT_CD    28 non-null     object 
 3   EVT_CL_CD      28 non-null     int64  
 4   RPTER_SEX      28 non-null     object 
 5   HPPN_PNU_ADDR  28 non-null     object 
 6   HPPN_X         27 non-null     float64
 7   HPPN_Y         27 non-null     float64
 8   SME_EVT_YN     22 non-null     object 
dtypes: float64(2), int64(1), object(6)
memory usage: 2.2+ KB


In [54]:
cn.RPTER_SEX.unique()

array(['남성', '여성', '불상'], dtype=object)

In [70]:
print(f'남성: {len(cn.loc[cn.RPTER_SEX == "남성"])}', end= " / ")
print(f'여성: {len(cn.loc[cn.RPTER_SEX == "여성"])}', end= " / ")
print(f'불상: {len(cn.loc[cn.RPTER_SEX == "불상"])}')

남성: 15083 / 여성: 13051 / 불상: 3461


In [72]:
total = cn.RPTER_SEX.count()
man = len(cn.loc[cn.RPTER_SEX == '남성'])
women = len(cn.loc[cn.RPTER_SEX == '여성'])
other = len(cn.loc[cn.RPTER_SEX == '불상'])

print(f"남성비율 : {(man/total) * 100}", end= " / ")
print(f"여성비율 : {(women/total) * 100}", end= " / ")
print(f"불상비율 : {(other/total) * 100}")

남성비율 : 47.738566228833676 / 여성비율 : 41.30716885583162 / 불상비율 : 10.954264915334704


충남청에 보이스피싱으로 신고된건이 많음. <br>
총 31595건 중 28건만 도착상태이고 나머지는 다 종결요청상태. <br>
전체에서 남자는 47.7%, 여자는 41.3% 이다.